fiftyone/factory/repos/execution_store.py

```diff
@@ -38,15 +33,46 @@ def create_store(self, store_name, permissions=None) -> StoreDocument:
         self._collection.insert_one(store_doc.to_mongo_dict())
         return store_doc
 
+    def has_store(self, store_name):
+        """Checks whether a store with the given name exists in the current
+        context.
+        """
+        result = self._collection.find_one(
+            dict(
+                store_name=store_name,
+                key="__store__",
+                dataset_id=self._dataset_id,
+            ),
+            {},
+        )
+        return bool(result)
+
     def list_stores(self) -> list[str]:
-        """Lists all stores in the execution store."""
-        # ensure that only store_name is returned, and only unique values
-        return self._collection.distinct("store_name")
+        """Lists the stores associated with the current context."""
+        result = self._collection.find(
+            dict(key="__store__", dataset_id=self._dataset_id),
+            {"store_name": 1},
+        )
+        return [d["store_name"] for d in result]
+
+    def count_stores(self) -> int:
+        """Counts the stores associated with the current context."""
+        return self._collection.count_documents(
+            dict(key="__store__", dataset_id=self._dataset_id),
+        )
+
+    def delete_store(self, store_name) -> int:
+        """Deletes the specified store."""
+        result = self._collection.delete_many(
+            dict(store_name=store_name, dataset_id=self._dataset_id)
```

Comment:
I think it is important to explicitly include `dataset_id: None` in this query (and all similar queries) when using an `ExecutionStoreService()` with no dataset context. As is, someone working with no dataset context could delete a store associated with a dataset.

In other words, I think there are `n + 1` possible contexts: one for each dataset, plus one additional "no dataset" context that is scoped to stores that are not associated with a dataset (which we are not currently using, but we could use soon to store app-wide state).

I added a  couple `XXX_global()` methods that really do operate across the entire execution store. But I think users should have to explicitly opt into those methods, rather than the current behavior where some methods are scoped to datasets but then other methods are "global" with `XXX_for_dataset()` methods that are scoped specifically to the dataset.

Reply:
> I think it is important to explicitly include dataset_id: None in this query

Yes... originally I was trying to avoid having a `dataset_id: null` in every document... however I agree with the explicitness, because `{key: 'blah'}` is an ambiguous query.

Reply:
I updated the class docstrings to explain what "context" means: https://github.com/voxel51/fiftyone/pull/5086/commits/07a515faa178c42c92542ca095f3affc129b9611